quil-rs/src/expression/mod.rs

```diff
@@ -848,6 +848,14 @@ mod tests {
         )
     }
 
+    macro_rules! prop_assert_eq_or_nan {
+        ($left:expr, $right:expr) => {
+            if !is_nan($left) && !is_nan($right) {
+                prop_assert_eq!($left, $right);
+            }
+        };
```

Comment:
This does nothing in the case where only one is `NaN`, which is incorrect.

```suggestion
        ($left:expr, $right:expr) => {
            if !is_nan($left) || !is_nan($right) {
                prop_assert_eq!($left, $right);
            }
        };
```

Reply:
If only one is NaN, my assumption is `prop_assert_eq!($left, $right);` would return false, which is what we want, right?